### PR TITLE
[WIP] Engineering 2.0: The Tesla Tragedy (Gnomes Ahoy)

### PR DESCRIPTION
**Pull request by Phatcat**
Nothing much to see here yet. That gnomish Death Ray sure is one hell of a messed up item spell-mechanics wise....

---

This change is [Reviewable](https://reviewable.io/reviews/cmangos/mangos-classic/138)

### CODE REVIEW

**Comment by Phatcat** [comment]
Wait..... gnomish universal control is channelling.

**Comment by Phatcat** [comment]
That's better

File: src/game/SpellEffects.cpp [hunk 1]
```diff
@@ -446,6 +446,27 @@ void Spell::EffectDummy(SpellEffectIndex eff_idx)
                     m_caster->CastSpell(m_caster, spell_id, true, nullptr);
                     return;
                 }
+                case 8344:                                  // Gnomish Universal Remote (ItemID: 7506)
+                {
+                    Unit* channelTarget = m_originalCaster->GetMap()->GetUnit(m_originalCaster->GetChannelObjectGuid());
```
Comment (VladimirMangos) [inline src/game/SpellEffects.cpp]: You need check m_originalCaster != null before use. In some cases GUID can't be resolved to pointer in time effect apply

Comment (Phatcat) [inline src/game/SpellEffects.cpp]: You're absolutely right. If you look at Death ray I have remembered it there, this was a last minute change and I forgot to add in the check after changing it from casting on unitTarget to cast on channelTarget.

**Comment by Phatcat** [comment]
Not at all abandoned this or given up on this at all, but I need to learn a bit more about the spell system because some of these dummy effects are more complicated that I initially thought they'd be.

**Comment by cyberium** [comment]
Some part alredy in core Thank @Phatcat.

**Pull request closed without merging**